packages/server/src/db/services.rs

```diff
@@ -182,17 +183,27 @@ impl_get_all!(
 );
 
 impl_get_all_filtered!(
-    fn_name = get_all_courses_by_name,
+    fn_name = get_courses_by_ids,
+    db_item = Course,
+    db_coll_name = "Courses",
+    filter_by = "_id",
+    filter_type = "$in"
+);
+
+impl_get_all_filtered!(
+    fn_name = get_courses_by_name,
     db_item = Course,
     db_coll_name = "Courses",
-    filter_name = "name"
+    filter_by = "name",
+    filter_type = "$regex"
 );
 
 impl_get_all_filtered!(
-    fn_name = get_all_courses_by_number,
+    fn_name = get_courses_by_number,
```

Comment:
I think this name is pretty confusing (especially when we have another service calls "get_courses_by_ids").
Maybe a more appropriate name would be "get_courses_filtered_by_number"?